src/main/java/com/google/devtools/build/lib/syntax/EvalUtils.java

```diff
@@ -319,7 +319,7 @@ public static boolean toBoolean(Object o) {
         for (Map.Entry<?, ?> entries : dict.entrySet()) {
           list.add(entries.getKey());
         }
-        return  ImmutableList.copyOf(list);
+        return ImmutableList.copyOf(list);
```

Comment:
Random driveby while skimming the PR & seeing the whitespace bump on this line: Convert this over to an `ImmutableList.builderWithExpectedSize()` also?

It presumably isn't getting called much if your numbers didn't complain about it too, but it's making an `ArrayList` only to have `ImmutableList` immediately copy it all.

Reply:
Good catch, @deadmoose! I'm gradually going through all of them, but before I'm trying to use either JMH or async-profiler to see the impact, so it's a bit slow :) I'll shoot a PR with this change as soon as I confirm in JMH that it makes things faster. Most likely your suggestion should improve things, since my previous testing showed that `ImmutableList.builder` + `build` is interestingly faster than `ArrayList.add`, but I'll double check.